python/src/robyn/modeling/feature_engineering.py

```diff
@@ -0,0 +1,207 @@
+
+        Returns:
+            pd.DataFrame: The rolling window data.
+        """
+        pass
+
+    def _calculate_media_cost_factor(self, dt_input_roll_wind: pd.DataFrame) -> pd.Series:
+        """
+        Calculate the media cost factor.
+
+        Args:
+            dt_input_roll_wind (pd.DataFrame): The rolling window data.
+
+        Returns:
+            pd.Series: The media cost factor.
+        """
+        pass
+
+    def _run_models(self, dt_input_roll_wind: pd.DataFrame, media_cost_factor: pd.Series) -> Dict[str, Any]:
+        """
+        Run models on the rolling window data.
+
+        Args:
+            dt_input_roll_wind (pd.DataFrame): The rolling window data.
+            media_cost_factor (pd.Series): The media cost factor.
+
+        Returns:
+            Dict[str, Any]: The model results, including fitted models, plots, and predictions.
+        """
+        pass
+
+    def _fit_spend_exposure(
+        self, dt_spend_mod_input: pd.DataFrame, media_cost_factor: float, paid_media_var: str
+    ) -> Dict[str, Any]:
+        """
+        Fit the spend exposure model.
+
+        Args:
+            dt_spend_mod_input (pd.DataFrame): The spend and exposure data.
+            media_cost_factor (float): The media cost factor.
+            paid_media_var (str): The name of the paid media variable.
+
+        Returns:
+            Dict[str, Any]: The model results, plot data, and predictions.
+        """
+        pass
+
+    def _prophet_decomposition(self, dt_transform: pd.DataFrame) -> pd.DataFrame:
+        """
+        Decompose the time series data using the Prophet model.
+
+        Args:
+            dt_transform (pd.DataFrame): The DataFrame containing the time series data to be decomposed.
+
+        Returns:
+            pd.DataFrame: The DataFrame with added columns for the decomposed components.
+        """
+        pass
+
+    def _set_holidays(self) -> pd.DataFrame:
+        """
+        Set up the holidays for the Prophet model.
+
+        Returns:
+            pd.DataFrame: The DataFrame with holidays.
+        """
+        pass
+
+    def _check_no_variance(self, dt_transform: pd.DataFrame) -> None:
+        """
+        Check for columns with no variance in the transformed data.
+
+        Args:
+            dt_transform (pd.DataFrame): The transformed data to check.
+        """
+        pass
+
+    @staticmethod
+    def hyper_names(adstock: str, all_media: List[str], all_vars: Optional[List[str]] = None) -> List[str]:
+        """
+        Get the names of all hyperparameters.
+
+        Args:
+            adstock (str): The type of adstock transformation.
+            all_media (List[str]): List of all media variables.
+            all_vars (Optional[List[str]], optional): List of all variables. Defaults to None.
+
+        Returns:
+            List[str]: The names of all hyperparameters.
+        """
+        pass
+
+    @staticmethod
+    def hyper_limits() -> pd.DataFrame:
```

Comment:
data.entities.Hyperparameters may be better place for this.

Reply:
removed as these are already available as static methods within HyperParameter #1043